pyproject.toml

```diff
@@ -0,0 +1,80 @@
+[build-system]
+requires = ["hatchling", "hatch-fancy-pypi-readme"]
+build-backend = "hatchling.build"
+
+[project]
+name = "httpx"
+description = "The next generation HTTP client."
+license = "BSD-3-Clause"
+requires-python = ">=3.7"
+authors = [
+    { name = "Tom Christie", email = "tom@tomchristie.com" },
+]
+classifiers = [
+    "Development Status :: 4 - Beta",
+    "Environment :: Web Environment",
+    "Framework :: AsyncIO",
+    "Framework :: Trio",
+    "Intended Audience :: Developers",
+    "License :: OSI Approved :: BSD License",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
+    "Topic :: Internet :: WWW/HTTP",
+]
+dependencies = [
+    "certifi",
+    "httpcore>=0.15.0,<0.16.0",
+    "rfc3986[idna2008]>=1.3,<2",
+    "sniffio",
+]
+dynamic = ["readme", "version"]
+
+[project.optional-dependencies]
+brotli = [
+    "brotli; platform_python_implementation == 'CPython'",
+    "brotlicffi; platform_python_implementation != 'CPython'",
+]
+cli = [
+    "click==8.*",
+    "pygments==2.*",
+    "rich>=10,<13",
+]
+http2 = [
+    "h2>=3,<5",
+]
+socks = [
+    "socksio==1.*",
+]
+
+[project.scripts]
+httpx = "httpx:main"
+
+[project.urls]
+Changelog = "https://github.com/encode/httpx/blob/master/CHANGELOG.md"
+Documentation = "https://www.python-httpx.org"
+Homepage = "https://github.com/encode/httpx"
+Source = "https://github.com/encode/httpx"
+
+[tool.hatch.version]
+path = "httpx/__version__.py"
+
+[tool.hatch.build.targets.sdist]
```

Comment:
Do we also need to specify a section for wheels?

I looked the Hatch docs and noticed it shows to define both here: https://hatch.pypa.io/latest/build/

Reply:
Nope no need because the defaults are working https://hatch.pypa.io/latest/plugins/builder/wheel/#default-file-selection

> it’s interesting that the scope of this PR is updating metadata only [...] doesn’t change any of the existing tooling we have around scripts/

Yup the modularity is by design! You can use just the build system, publish wheels and source distributions that were built by other tools, only use the environment management, etc.

Speaking of environment management, in another PR I could switch those scripts to Hatch envs + [scripts](https://hatch.pypa.io/latest/config/environment/overview/#scripts) to better manage dev dependencies and so us Windows users could run them too 😉